src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/EggFinder.java

```diff
@@ -0,0 +1,139 @@
+package de.hysky.skyblocker.skyblock.chocolatefactory;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.ColorUtils;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.ItemUtils;
+import de.hysky.skyblocker.utils.SkyblockTime;
+import de.hysky.skyblocker.utils.waypoint.Waypoint;
+import it.unimi.dsi.fastutil.objects.ObjectImmutableList;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayConnectionEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.decoration.ArmorStandEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EggFinder {
+	private static final Pattern eggFoundPattern = Pattern.compile("^(?:HOPPITY'S HUNT You found a Chocolate|You have already collected this Chocolate) (Breakfast|Lunch|Dinner)");
+	private static final Pattern newEggPattern = Pattern.compile("^HOPPITY'S HUNT A Chocolate (Breakfast|Lunch|Dinner) Egg has appeared!$");
+	private static final Logger logger = LoggerFactory.getLogger("Skyblocker Egg Finder");
+
+	private EggFinder() {
+	}
+
+	public static void init() {
+		ClientPlayConnectionEvents.JOIN.register((ignored, ignored2, ignored3) -> clearEggs());
+		ClientReceiveMessageEvents.GAME.register(EggFinder::onChatMessage);
+		WorldRenderEvents.AFTER_TRANSLUCENT.register(EggFinder::renderWaypoints);
+	}
+
+	public static void checkIfEgg(Entity entity) {
+		if (entity instanceof ArmorStandEntity armorStand) checkIfEgg(armorStand);
+	}
+
+	public static void checkIfEgg(ArmorStandEntity armorStand) {
+		if (!SkyblockerConfigManager.get().helpers.chocolateFactory.enableEggFinder) return;
+		if (SkyblockTime.skyblockSeason.get() != SkyblockTime.Season.SPRING) return;
```

Comment:
While checking for the season is great it would also be good if we can also only check on islands where eggs can actually spawn, because checking in dungeons for example is not useful.